### PR TITLE
Messages Page

### PR DESCRIPTION
**Pull request by popkinj**
- New Messages tab
Screenshot from 2022-10-25 12-15-31
- New Messages page with table
Screenshot from 2022-10-25 12-15-56
- Custom API call
Screenshot from 2022-10-25 12-16-34

### CODE REVIEW

File: services/tenant-ui/frontend/src/components/contacts/Contacts.vue [hunk 1]
```diff
@@ -89,6 +89,7 @@ const toast = useToast();
 const contactsStore = useContactsStore();
 
 const { loading, contacts, selectedContact } = storeToRefs(useContactsStore());
+console.log('contact data', contacts);
```
Comment (loneil) [inline services/tenant-ui/frontend/src/components/contacts/Contacts.vue:92]: Extra console log

File: services/tenant-ui/frontend/src/store/index.ts [hunk 2]
```diff
@@ -7,6 +7,8 @@ export { useIssuerStore } from './issuerStore';
 export { useTenantStore } from './tenantStore';
 export { useTokenStore } from './tokenStore';
 export { useVerifierStore } from './verifierStore';
+export { useMessageStore } from './messageStore';
+``;
```
Comment (loneil) [inline services/tenant-ui/frontend/src/store/index.ts:11]: Blank quote here

File: services/tenant-ui/frontend/src/store/messageStore.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  // axios
+  //   .get('/tenant/v1/messages/')
+  //   .then((res) => {
+  //     selectedMessage.value = null;
+  //     console.log(res);
+  //   })
+  //   .catch((err) => {
+  //     console.log('yo mama');
+  //     console.log(err);
+  //   });
```
Comment (loneil) [inline services/tenant-ui/frontend/src/store/messageStore.ts:27]: Commented out code

**Review by loneil — CHANGES_REQUESTED** [review]
Created and recieved message to/from bc wallet connection. But row expander is on the table but not doing anything

image

**Comment by popkinj** [comment]
> Created and recieved message to/from bc wallet connection. But row expander is on the table but not doing anything
> 
> image

Was it supposed to do something?

File: services/tenant-ui/frontend/src/store/messageStore.ts (hunk 3 above)
Comment (popkinj) [reply services/tenant-ui/frontend/src/store/messageStore.ts]: Good catch.

File: services/tenant-ui/frontend/src/store/index.ts (hunk 2 above)
Comment (popkinj) [reply services/tenant-ui/frontend/src/store/index.ts]: Got it.

**Comment by loneil** [comment]
> > Created and recieved message to/from bc wallet connection. But row expander is on the table but not doing anything
> > image
> 
> I think it's hooked up... there's just no content to show.

It's not hooked up, it's missing the row expansion slot and the RowExpandData component that shows the GET result of the message record.

**Comment by popkinj** [comment]
> > > Created and recieved message to/from bc wallet connection. But row expander is on the table but not doing anything
> > > image
> > 
> > 
> > I think it's hooked up... there's just no content to show.
> 
> It's not hooked up, it's missing the row expansion slot and the RowExpandData component that shows the GET result of the message record.

k... I'll have another look.

**Comment by popkinj** [comment]
Screenshot from 2022-10-27 09-57-35
Think this is working now.

**Review by loneil — APPROVED** [review]
Working as expected. The Content table field doesn't make too much sense to have in there if Store Messages is off (the default) but we can look at that in a final UX later.

**Pull request merged**